codeready-toolchain/member-operator
# feat: migrate member to operator-sdk 1.8.0

## Migrates member-operator repo to operator-sdk 1.8.0

* adds kubebuilder's `PROJECT` file with the definition of the operator
* adds `budle/` & ` bundle.Dockerfile` to `.gitignore` since they are generated dynamically 
* cleans README from obsolete parts
* updates CRDs with the latest changes
* updates dependencies to use k8s v0.20.2 - as a result:
    *  updates the mappers used in controllers to match the latest API - it doesn't expect a type, but a function
    * migrates most of the code from `runtime.Object` to `client.Object`, but here is still for an improvement - will be done in next PR
    * use the new API for defining the watched resources in controllers: `ctrl.NewControllerManagedBy`
    * changes `Reconcile(request ctrl.Request)` to `Reconcile(ctx context.Context, request ctrl.Request)`
* updated go to 1.16
    * in go.mod
    * in GH Actions workflows  
* don't store log inside the `Reconciler` type but create it using `log.FromContext(ctx)`
* start using `GetWatchNamespace` & `GetOperatorName` from toolchain-common
* deletes `hack/` directory - it's not needed anymore
* as for the operator's `main.go` file
    * it was moved to the root of the repo
    * updates the content to get as close as possible to the version generated by latest operator-sdk/kubebuilder
* as for make targets
    * drops `dev.mk` as it's not needed anymore (or at least for now)
    * gets `kustomize` binary via `get-tool.mk` so it can be used when generating operator manifests
    * when generating assets it uses `go-bindata` binary fetched via `get-tool.mk`  as well
    * significant changes in `manifests.mk` file to match the latest changes in CI/CD setup/logic
    * also updates `test.mk` so it takes into account also the fact that it is executed in GH Action workflow
* as for Dockerfile.tools
    * updated go to 1.16
    * updated oc to the latest 4.7
    * dropped operator-sdk as is not needed in openshift-ci anymore

plus:
* changes the name of the index image to `member-operator-index` ` (for better isolation)

### TODO in next PRs
* remove OLM related files and directories from `deploy/` folder
* add CRD samples
* investigate scorecard tests
* clean openshift/release config for openshift-ci from obsolete parts
